### ISSUE TITLE
[Bug]: DataTable toolbar with overflow menu - options icon not visible in Storybook

### ISSUE THREAD

**Issue by warrenmblood** [carbon-design-system/carbon#19281]
### Package

@carbon/web-components

### Browser

Chrome

### Package version

1.81.0

### Description

In the DataTable toolbar with overflow menu storybook, the options menu icons are not visible and appear white on hover, resulting in very low contrast with background:

![Image](https://github.com/user-attachments/assets/cd78ad33-ce4e-4fa8-b160-36923bfa1c09)

### Reproduction/example

https://web-components.carbondesignsystem.com/?path=/story/components-datatable-toolbar--with-overflow-menu

### Steps to reproduce

1. Go to https://web-components.carbondesignsystem.com/?path=/story/components-datatable-toolbar--with-overflow-menu
2. Observe the options menu icons on the right end of each table row with/without hovering

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
